packages/headless/src/features/tab-set/tab-set-slice.ts

```diff
@@ -0,0 +1,32 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {registerTab, updateActiveTab} from './tab-set-actions';
+import {getTabSetInitialState} from './tab-set-state';
+
+export const tabSetReducer = createReducer(
+  getTabSetInitialState(),
+  (builder) => {
+    builder
+      .addCase(registerTab, (state, action) => {
+        const tab = action.payload;
+        const {id} = tab;
+
+        if (id in state) {
```

Comment:
Should this be a warning of some sort ? 
Instead of silent return.

Reply:
I believe a silent return is the correct behavior at the reducer level, especially for `register` actions. Here's why,

Noop register actions are a feature of headless since day one. They make it possible to synchronize different controller instances, without overwriting state.

Two scenarios for controller synchronization:
1. Server-side rendering. When the client is hydrating, controllers are instantiating. However, the ids already exist in state which was created on the server.
2. If someone wants to act on a particular instance in a `Set` from different UI components, they can simply instantiate a controller on the same id. If this was not possible, one would have to share the controller instance some how e.g. by passing it as a prop, using a global etc.

So, while entering this `if` can be a mistake, it is not necessarily one, and often will be intentional. Logging a warning would often just be adding spam the console.